bindings/java/src/org/sleuthkit/datamodel/OsAccountRealmManager.java

```diff
@@ -220,51 +220,134 @@ public OsAccountRealm createWindowsRealm(String accountSid, String realmName, Ho
-			return updateRealm(realm, connection);
+			return updateRealm(realm, realmAddr, realmName, connection);
 		}
 	}
 		
 	/**
-	 * Updates the specified realm in the database.
-	 * 
-	 * @param realm Realm to update.
+	 * Updates the address and/or name of specified realm in the database.
+	 *
+	 * Updates the realm address and/or name, if a non blank address/name is
+	 * specified and the current address/name is blank.
+	 *
+	 * @param realm      Realm to update.
+	 * @param realmAddr  Realm address, may be null if the address doesn't need
+	 *                   to be updated.
+	 * @param realmName  Realm name, may be null if the name doesn't need to be
+	 *                   updated.
 	 * @param connection Current database connection.
-	 * 
-	 * @return OsAccountRealm Updated realm.
-	 * 
-	 * @throws TskCoreException 
+	 *
+	 * @return RealmUpdateStatus Update status and updated realm.
+	 *
+	 * @throws TskCoreException If there is a database error or if a realm
+	 *                          already exists with that information.
 	 */
-	private OsAccountRealm updateRealm(OsAccountRealm realm, CaseDbConnection connection) throws TskCoreException {
+	private RealmUpdateStatus updateRealm(OsAccountRealm realm, String realmAddr, String realmName,  CaseDbConnection connection) throws TskCoreException {
 		
-		// do nothing if the realm is not dirty.
-		if (!realm.isDirty()) {
-			return realm;
-		}		
+		// need at least one of the two
+		if (StringUtils.isBlank(realmAddr) && StringUtils.isBlank(realmName)) {
+			throw new TskCoreException("Realm address or name is required to update realm.");
+		}
+		
+		RealmUpdateStatusEnum updateStatusCode = RealmUpdateStatusEnum.NO_CHANGE;
+		OsAccountRealm updatedRealm = null;
 		
 		List<String> realmNames = realm.getRealmNames();
-		String realmName = realmNames.isEmpty() ? null : realmNames.get(0);
-			
+		String currRealmName = realmNames.isEmpty() ? null : realmNames.get(0);	// currently there is only one name.
+		String currRealmAddr = realm.getRealmAddr().orElse(null);
+		
+		
+		// set name and address to new values only if the current value is blank and the new value isn't.		
+		String newRealmAddr;
+		if ( (StringUtils.isBlank(currRealmAddr) && StringUtils.isNotBlank(realmAddr))) {
+			newRealmAddr = realmAddr;
+			updateStatusCode = RealmUpdateStatusEnum.UPDATED;
+		} else {
+			newRealmAddr = currRealmAddr;
+		}
+		
+		String newRealmName;
+		if (StringUtils.isBlank(currRealmName) && StringUtils.isNotBlank(realmName)) {
+			newRealmName = realmName;
+			updateStatusCode = RealmUpdateStatusEnum.UPDATED;
+		} else {
+			newRealmName = currRealmName;
+		}
+		
+		// make new signature
+		String newSignature = makeRealmSignature(newRealmAddr, newRealmName, realm.getScopeHost().orElse(null));
+		
+		// if nothing is to be changed, return
+		if ( updateStatusCode == RealmUpdateStatusEnum.NO_CHANGE) {
+			return new RealmUpdateStatus(updateStatusCode, null);
```

Comment:
I believe we need to always return an OsAccountRealm here, even if it is the one passed in.